### PR TITLE
Minor changes to indigo-devel CMake allow this to be used in kinetic and indigo

### PR DESCRIPTION
**Pull request by pbeeson**
I still needed this package, despite the fact that its being released via the Debian packages anymore.  I forked and made the simple changes in the CMakefile that allow this to run in 14.04 or 16.04 with Indigo, Jade, and Kinetic.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,11 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(robot_self_filter)
 
+if(CMAKE_COMPILER_IS_GNUCXX)
+    set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
```
Comment (wkentaro) [inline CMakeLists.txt:6]: Could you please explain why this is needed on kinetic?

**Comment by pbeeson** [comment]
I'm using kinetic and I need it???

Why would it not be needed on kinetic?

Is it folded into another package?

**Comment by pbeeson** [comment]
Maybe I read your comment incorrectly. This particular fix is needed under
Kinetic because much of kinetic now uses C++-11 or higher. I'm not sure
exactly why, the you package doesn't compile under Kinetic without
explicitly stating C++-11 is needed. I've seen this in several other
packages as well.

**Comment by pbeeson** [comment]
For example see https://github.com/ros-industrial/industrial_core/pull/161

**Comment by wkentaro** [comment]
I wonder why this is not addressed on core packages like `cv_bridge` and `nodelet`, which are already released to kinetic. Do you know why?
https://github.com/ros-perception/vision_opencv/blob/kinetic/cv_bridge/CMakeLists.txt
https://github.com/ros/nodelet_core/blob/indigo-devel/nodelet/CMakeLists.txt

Or are there any information on roswiki?
I could not find it on the migration page: http://wiki.ros.org/kinetic/Migration

**Comment by pbeeson** [comment]
The newest compilers on 16.04 require c++-11 and higher to be explicitly
set for certain things. I don't recall what the exact compiler error that
made me figure out the c++-11 was needed, but it shouldn't be hard to
reproduce. I'll be back at my desk on Tuesday and can try then.

**Comment by pbeeson** [comment]
Is this able to be merged and resolved?  It would be nice to have this percolate through the 14.04 and 16.04 build farm and not have to keep a fork of your repo around anymore.

**Pull request merged**